src/hotspot/share/memory/metadataFactory.hpp

```diff
@@ -61,14 +62,21 @@ class MetadataFactory : AllStatic {
 
   // Deallocation method for metadata
   template <class T>
-  static void free_metadata(ClassLoaderData* loader_data, T md) {
+  static void free_metadata(ClassLoaderData* loader_data, T* md) {
     if (md != NULL) {
       assert(loader_data != NULL, "shouldn't pass null");
       int size = md->size();
-      // Call metadata's deallocate function which will call deallocate fields
+      // Call metadata's deallocate function which will deallocate fields and release_C_heap_structures
       assert(!md->on_stack(), "can't deallocate things on stack");
       assert(!md->is_shared(), "cannot deallocate if in shared spaces");
       md->deallocate_contents(loader_data);
+      // Call the destructor. This is currently used for MethodData which has a member
+      // that needs to be destructed to release resources. Most Metadata derived classes have noop
+      // destructors and/or cleanup using deallocate_contents.
+      // T is a potentially const or volatile qualified pointer. Remove any const
```

Comment:
We'll given the template, T by itself is no longer a pointer so its just potentially const or volatile qualified. But that is pedantic.

Reply:
I just want to leave it because you're right but pointer is a readable noun here.